controllers/kafkamanagement/kafkauser_controller.go

```diff
@@ -533,3 +551,295 @@ func (r *KafkaUserReconciler) getKafkaUserCredsFromSecret(
+) {
+	ctx := context.TODO()
+	l := log.FromContext(ctx)
+
+	for _, oldCert := range oldCerts {
+		var exist bool
+		for _, newCert := range newObj.Spec.CertificateCreationData {
+			if *oldCert == *newCert {
+				exist = true
+				break
+			}
+		}
+
+		if exist {
+			continue
+		}
+
+		err := r.handleDeleteCertificate(ctx, newObj, l, oldCert)
+		if err != nil {
+			l.Error(err, "Cannot delete Kafka user from the cluster", "user", oldCert)
+			r.EventRecorder.Eventf(newObj, models.Warning, models.CreatingEvent,
+				"Cannot delete user from the cluster. Reason: %v", err)
+		}
+	}
+
+	for _, newCert := range newObj.Spec.CertificateCreationData {
+		var exist bool
+		for _, oldCert := range oldCerts {
+			if *newCert == *oldCert {
+				exist = true
+				break
+			}
+		}
+
+		if exist {
+			continue
+		}
+
+		err := r.handleCreateCertificate(ctx, newObj, l, newCert)
+		if err != nil {
+			l.Error(err, "Cannot create Kafka user Mtls certificate", "cert", newCert)
+			r.EventRecorder.Eventf(newObj, models.Warning, models.CreatingEvent,
+				"Cannot create user Mtls certificate. Reason: %v", err)
+		}
+
+		oldCerts = append(oldCerts, newCert)
+	}
+}
+
+func (r *KafkaUserReconciler) handleCreateCertificate(ctx context.Context, user *v1beta1.KafkaUser, l logr.Logger, certCreationData *v1beta1.CertificateCreationData) error {
+	username, _, err := r.getKafkaUserCredsFromSecret(user.Spec)
+	if err != nil {
+		l.Error(
+			err, "Cannot get Kafka user creds from secret",
+			"kafka user spec", user.Spec,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.FetchFailed,
+			"Fetch user credentials from secret is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	if certCreationData.CSR == "" {
+		certCreationData.CSR, err = r.GenerateCSR(certCreationData)
+		if err != nil {
+			l.Error(err, "Cannot generate CSR for Kafka user certificate creation",
+				"user", user.Name,
+			)
+			r.EventRecorder.Eventf(
+				user, models.Warning, models.GenerateFailed,
+				"Generate CSR is failed. Reason: %v",
+				err,
+			)
+
+			return err
+		}
+	}
+
+	certResponse, err := r.API.CreateKafkaUserCertificate(instaclustr.KafkauserCertificatesEndpoint, certCreationData, username)
+	if err != nil {
+		l.Error(err, "Cannot create Kafka user Mtls certificate",
+			"user", user.Name,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.CreationFailed,
+			"Certificate creation is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
+
+	newSecret := user.NewCertificateSecret(certCreationData.SecretName, certCreationData.SecretNamespace)
+	err = r.Client.Create(ctx, newSecret)
+	if k8serrors.IsAlreadyExists(err) {
+		l.Error(err, "Cannot create Kafka user Cert Secret. Secret already exist",
+			"secret name", certCreationData.SecretName,
+		)
+		r.EventRecorder.Eventf(
+			user, models.Warning, models.CreationFailed,
+			"create user Cert Secret is failed. Reason: %v",
+			err,
+		)
+
+		return err
+	}
```

Comment:
Here you handle only `IsAlreadyExists` error and skip others. Please handle generic error.
```suggestion
	if err != nil {
	    if k8serrors.IsAlreadyExists(err) {
	        l.Error(err, "Cannot create Kafka user Cert Secret. Secret already exist",
			    "secret name", certCreationData.SecretName,
		    )
		    r.EventRecorder.Eventf(
			    user, models.Warning, models.CreationFailed,
		      	"create user Cert Secret is failed, secret already exists. Reason: %v",
			    err,
		    )

		    return err
	    }
	    
		l.Error(err, "Cannot create Kafka user Cert Secret.",
			"secret name", certCreationData.SecretName,
		)
		r.EventRecorder.Eventf(
			user, models.Warning, models.CreationFailed,
			"create user Cert Secret is failed. Reason: %v",
			err,
		)

		return err
	}
```

Reply:
done